### DISCUSSION TITLE
About the microphone issues in QtWebengine

### DISCUSSION THREAD

**Discussion by ManuGraiph** [qutebrowser/qutebrowser#7737] [Q&A]
Hi..

I have the same issue that has been reported before of the mic not being recognized by qutebrowser while using pipewire due to some issues in QtWebengine.

I'd like to ask if there is some solution for that, other than switching to pulseaudio (in arch), cause pretty much it's the only thing keeping me from switching to qutebrowser.

**Comment by toofar** [comment]
> I have the same issue that has been reported before of the mic not being recognized by qutebrowser while using pipewire due to some issues in QtWebengine.

Can you link the issue/comment please? Or add more information about your own issue? It seems to work well enough for me. I've been using pipewire for several months now and switched from alsa -> pipewire to the pipewire-pulse daemon recently and both configurations have worked fine.

![image](https://github.com/qutebrowser/qutebrowser/assets/7419144/d1acf847-05df-440a-8bf8-0b2cd84eda9c)

Look I even took a picture!
(1) the input node into qutebrowser (sadly it's not picking up the right appname or icon) via qpwgraph
(2) the same thing in easyeffects
(3) me yelling

This is with pipewire 0.3.65 and wireplumber 0.4.13 (I don't think the version matters though) and easyeffects running (without default output selected) and the pipewire-pulse daemon running (which is what webengine is talking to, this doesn't require pulse to be running, or installed).

**Comment by ManuGraiph** [comment]
I have exactly that situation, with pipewire only output works.

https://github.com/qutebrowser/qutebrowser/issues/7053

**Reply by toofar** [reply]
It seems like an issue with your setup, so you are going to have to do a bit more debugging. Some ideas:

* does the error come before or after the permission request? (I assume `content.media.audio_capture` is set to "ask", which is the default)
* test in a temp basedir (eg `qutebrowser -T onlinemictest.com`)
* run pw-mon in a terminal and see if there is any error printed out to there
* launch qutebrowser with the chrome [debug logging](https://support.google.com/chrome/a/answer/6271282?hl=en#zippy=%2Clinux) turned on. Eg `qutebrowser -T -s qt.args '["enable-logging", "v=1"]' onlinemictest.com | tee lotsoflogs.txt` and see if there are any errors in there, particulalry around audio, media or input.
* try running against Qt6 and see if it works any better

**Comment by ManuGraiph** [comment]
An "issue with my setup" where other people have the same issue and the main dev of qb also confirmed? Did you see the issue i linked?

**Reply by The-Compiler** [reply]
I can't really see me confirm anything there - I never tried to reproduce, and it looks like you never got the error mentioned in that issue?
